PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/configuration/LivySparkBatchJobRunConfiguration.java

```diff
@@ -270,8 +271,14 @@ public void checkRunnerSettings(@NotNull ProgramRunner runner, @Nullable RunnerS
         super.checkRunnerSettings(runner, runnerSettings, configurationPerRunnerSettings);
     }
 
-    public Observable<ISparkBatchJob> buildSparkJobForRunner(SparkSubmissionRunner runner) {
-        return runner.buildSparkBatchJob(getSubmitModel(), PublishSubject.create());
+    @Override
+    public Observable<ISparkBatchJob> prepare(final ProgramRunner<RunnerSettings> runner) {
+        if (!(runner instanceof SparkSubmissionRunner)) {
+            return Observable.empty();
```

Comment:
I'm not sure about this suggestion but do we need to throw a exception or add some logs here rather than return an empty observable?

Reply:
En, should assert it here.